src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java

```diff
@@ -1811,9 +1826,13 @@ private static String parameterizedTypeToString(final ParameterizedType p) {
         return buf.toString();
     }
 
-    private static void appendRecursiveTypes(final StringBuilder buf, final int[] recursiveTypeIndexes, final Type[] argumentTypes) {
-        for (int i = 0; i < recursiveTypeIndexes.length; i++) {
-            appendAllTo(buf.append('<'), ", ", argumentTypes[i].toString()).append('>');
+    @SuppressWarnings({"compound.assignment.type.incompatible","array.access.unsafe.high","assignment.type.incompatible"})/*
```

Comment:
Given your justifications, the SameLen annotation here should make this code typecheck. Why doesn't it?